tests/data/policy/agent_policy.rego

```diff
@@ -21,62 +21,50 @@ import rego.v1
 
 policy_version := "0.1.0"
 
+# Defines default rules to be used if not specified in policy file.
 default default_global_rules := {
-	"allowListOnly": false,
+	"allowListOnly": true,
 	"signingRules": {
-		"extensionSigned": false,
-		"signingDetails": {},
-	},
-	"updateAllowed": true,
-	"uninstallAllowed": true,
+		"extensionSigned": true
+	}
 }
 
 default global_rules := {
-	"allowListOnly": false,
+	"allowListOnly": true,
 	"signingRules": {
-		"extensionSigned": false,
-		"signingDetails": {},
-	},
-	"updateAllowed": true,
-	"uninstallAllowed": true,
+		"extensionSigned": true
+	}
 }
 
+# Rules from policy file (data.azureGuestAgentPolicy) override the default rules if present.
 global_rules := object.union(default_global_rules, data.azureGuestAgentPolicy) if {
 	data.azureGuestAgentPolicy
 }
 
 default any_extension_allowed := true
 
+# If allowlist rule is true, download is not allowed by default.
 any_extension_allowed := false if {
 	global_rules.allowListOnly
 }
 
 default default_signing_info := {"signingInfo": {}}
 
-# Download rule 1: if the extension is in the list and download rule satisfied: download allowed
+# Download rule 1: if extension is in the allowlist, download is allowed.
 extensions_to_download[name] := extension if {
 	some name, input_extension in input.extensions
 	data.azureGuestExtensionsPolicy[name]
```

Comment:
+1; also pointed out int he spec review

Reply:
Eventually, if we wanted to add other non-extension policies to the single data file, it would go under the "azureGuestAgentPolicy" section (for example, JIT policy). So for now it's extensions only, but it will contain all policies globally affecting agent behavior. 

We could change it to "azureGuestAgentGlobalPolicy" and "azureGuestExtensionsIndividualPolicy", if that's clearer.

Reply:
That naming makes more sense to me, but we should still update the properties in the global policy section to indicate they are related to extensions. 
'allowListOnly' isn't clear that this property is related to extensions for example

Reply:
How about going from

```
{
   "azureGuestAgentPolicy": {
      "policyVersion": "0.1.0",
      "signingRules": {
         "extensionSigned": true
      },
      "allowListOnly": true
   },
   "azureGuestExtensionsPolicy": {
      "<extension name1>": {
         "runtimeRules": {
            "filesMustBeSigned": true,
            "fileRootCert": "<dirs>",
            "allowedCommands": ["<cmd1>, "<cmd2>" ]
         }
      }
   }
}

```
to something similar to

```
{
   "policyVersion": "0.1.0",
   "extensionPolicies": {
      "allowAllExtensions": false,
      "allowUnsignedExtensions": false
   },
   "<extension name1>": {
      "allowUnsigned": true,
      "fileRootCert": "<dirs>",
      "allowedCommands": [ "<cmd1>, "<cmd2>" ]
   }
}
```

Reply:
I put signing related rules under the tag of "signingRules" because we could add additional signing validation rules in the future, for example certain certificate validation policy: (what time to use: current time or signing time, whether we validate certificate usage, etc. All these options were valid options when validating signature and certificate, but until we received requests from customers), we don't need to change the overall structure of Rego, rather changing the one validation function we have at the end of this rego file. Similarly, I put "fileRootCert", "allowedcommands" under "runtime_rules" because each extension may have its own settings which the agent rego rule doesn't need to understand. I was even thinking customer could encrypt these settings and put encryped string under the runtime settings, like how environment variables being passed to extension. For the detailed word choices of the tag names like allowAllExtensions and allowUnsignedExtensions , it may make the rules look more complicated. I need to experiment.  but it does look better. if they don't increase the complexity, I would like to follow the suggestion.